webapp/vue.md

```diff
@@ -0,0 +1,53 @@
+# Vue – Code Guidelines
+
+## We use single-file components
+
+Each component lives in a single file, containing:
+- its `template` (the DOM structure)
+- its `script` (including `props`, `data` and `methods` among other things)
+- its `style` (defining the look of the component)
+
+See the [Vue.js docs](https://vuejs.org/v2/guide/single-file-components.html) for more details.
+
+Placed in the same folder are also:
+- the test file (e.g. `MyComponent.spec.js`)
+- the storybook file (e.g. `MyComponent.story.js`)
+
+## We use typed props
+
+Vue.js allows us to define component props either as strings or as objects (with `type` and `default` or `required` values). Always go for the second option!
+
+Also: only (and always!) define a `default` for props that are _not required_.
+
+Why?
+- it makes our code more robust – a warning will be shown when passing a wrong prop type
+- it clearly defines the component API and tells other developers how to use it
+
+It is as easy as writing:
+
+```
+props: {
+  title: {
+    type: String,
+    required: true,
+  },
+  image: {
+    type: String,
+    default: 'human-connection-logo.png',
+  },
+}
+```
+
+For more complex use cases see the [official Vue.js documentation](https://vuejs.org/v2/guide/components-props.html#Prop-Validation).
+
+## We use shorthands
+
+For better readability we prefer
+- `:something` over `v-bind:something`
+- `@click` over `v-on:click`
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-31T11:12:30Z" title="Thursday, October 31st 2019, 12:12:30 pm +01:00">Oct 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,53 @@
+# Vue – Code Guidelines
+
+## We use single-file components
+
+Each component lives in a single file, containing:
+- its `template` (the DOM structure)
+- its `script` (including `props`, `data` and `methods` among other things)
+- its `style` (defining the look of the component)
+
+See the [Vue.js docs](https://vuejs.org/v2/guide/single-file-components.html) for more details.
+
+Placed in the same folder are also:
+- the test file (e.g. `MyComponent.spec.js`)
+- the storybook file (e.g. `MyComponent.story.js`)
+
+## We use typed props
+
+Vue.js allows us to define component props either as strings or as objects with `type`, `default` and `required` values. Always go for the second option!
+
+Also: define defaults _only and always_ for non-required props.
+
+Why?
+- it makes our code more robust – a warning will be shown when passing a wrong prop type
+- it clearly defines the component API and tells other developers how to use it
+
+It is as easy as writing:
+
+```
+props: {
+  title: {
+    type: String,
+    required: true,
+  },
+  image: {
+    type: String,
+    default: 'human-connection-logo.png',
+  },
+}
+```
+
+For more complex use cases see the [official Vue.js documentation](https://vuejs.org/v2/guide/components-props.html#Prop-Validation).
+
+## We use shorthands
+
+For better readability we prefer
+- `:something` over `v-bind:something`
+- `@click` over `v-on:click`
```


nice to have this documented!